### DISCUSSION TITLE
Can someone explain zoom levels, maxZoomPixelRatio and zoomInMultiplier options?

### DISCUSSION THREAD

**Discussion by Andrew-web-coder** [igordanchenko/yet-another-react-lightbox#314] [Q&A] [answered]
Sorry, but I just do not understand what these options do.

> maxZoomPixelRatio - ratio of image pixels to physical pixels at maximum zoom level

What is the ratio, of what? And why? Can someone, please, explain a bit?

> zoomInMultiplier - zoom-in multiplier

What does it multiply and why? 

And also, I do not understand how zoom in/out buttons determine what zoom level to set? I think it does not show image in the actual size. Maybe I am missing something? Also, I find it confusing that clicking the image zooms with different zoom levels than buttons in the toolbar :/

**Comment by igordanchenko** [comment, marked as answer]
>> maxZoomPixelRatio - ratio of image pixels to physical pixels at maximum zoom level
>
> What is the ratio, of what? And why? Can someone, please, explain a bit?

This prop determines the maximum zoom level, or in other words, how far you can zoom in on a picture. 

`rendered image dimensions at maximum zoom` = `intrinsic image dimensions ` * `maxZoomPixelRatio`

The default value is `1`, which avoids image pixelization at maximum zoom.

>> zoomInMultiplier - zoom-in multiplier
>
> What does it multiply and why?

This prop determines the degree to which the zoom level changes when you click the zoom-in / zoom-out buttons.

`new zoom value` = `zoom value` * `zoomInMultiplier`

> And also, I do not understand how zoom in/out buttons determine what zoom level to set? 

See above.

> I think it does not show image in the actual size. Maybe I am missing something?

Most likely.

> Also, I find it confusing that clicking the image zooms with different zoom levels than buttons in the toolbar :/

Double-click / double-tap zoom gestures use a different algorithm. With double-click / double-tap, you can go from minimum to maximum zoom level in no more than just two clicks.

**Reply by Andrew-web-coder** [reply]
Thanks! I wonder why it is named `maxZoomPixelRatio` and not smth like `maxZoomFactor`, I was confused, because I thought it is somehow related to `window.devicePixelRatio`.

**Reply by igordanchenko** [reply]
There's usually more than one way to name the same thing. Even `zoomOomph` could be a descriptive name for this prop.
